PYIC-7329: Refactor core-front tests

MikeCollingwood opened the pull request:
## Proposed changes

### What changed

- Standardise core-front tests
- Use mock-express test util to commonise how we defined req & res in core-front tests
- Use `sinon.fake`'s wherever possible as is the simplest class of mock
- Inline variables wherever sensible, to be consistent and it is more concise.
- Added in Arrange, Act and Assert comments to help readability

### Why did it change

- Improve readability & maintainability

### Issue tracking

- [PYIC-7329](https://govukverify.atlassian.net/browse/PYIC-7329)

[PYIC-7329]: https://govukverify.atlassian.net/browse/PYIC-7329?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

MikeCollingwood committed:
PYIC-7329: Refactor core-front tests

MikeCollingwood committed:
PYIC-7329: Fix tests for rebase

MikeCollingwood commented on src/app/ipv/tests/checkFormRadioButtons.test.ts:
```diff
@@ -1,39 +1,37 @@
+  // Mock handler parameters
+  const createRequest = specifyCreateRequest({
     params: { pageId: "page-ipv-identity-document-start" },
-    csrfToken: sinon.fake(),
```
Does not need these defined as default in specifyCreateRequest

MikeCollingwood committed:
PYIC-7329: Change .reset() to resetHistory for safety

MikeCollingwood commented on src/app/credential-issuer/middleware.ts:
```diff
@@ -44,7 +44,7 @@ export const sendParamsToAPI: RequestHandler = async (req, res) => {
 
   const apiResponse = await postCriCallback(req, body);
-  return handleBackendResponse(req, res, apiResponse?.data);
+  return handleBackendResponse(req, res, apiResponse.data);
```
`data` is not an optional parameter on AxiosResponse, the return type of postCriCallback

MikeCollingwood committed:
PYIC-7329: Move repeated stub resolve to outside beforeEach

MikeCollingwood committed:
PYIC-7329: Move towards using fake over stub for simplicity

MikeCollingwood committed:
PYIC-7329: Use nullish check instead of logical ORs

MikeCollingwood committed:
PYIC-7329: Remove sonarcloud analysis of router.ts

DanCorderIPV commented on src/test-utils/mock-express.ts:
```diff
@@ -0,0 +1,70 @@
+}
+
+export const specifyCreateRequest =
+  (commonParams: CreateRequestParams = {}) =>
```
I think calling this `defaultParams` might make it a little easier to understand

DanCorderIPV commented on src/test-utils/mock-express.ts:
```diff
@@ -0,0 +1,70 @@
+
+export const specifyCreateRequest =
+  (commonParams: CreateRequestParams = {}) =>
+  (params: CreateRequestParams = {}): Request => {
```
This is the linter's fault, but I really don't like how it's arranged this code :(

DanCorderIPV commented on src/services/parameterStoreService.test.ts:
```diff
@@ -56,18 +64,20 @@ describe("parameterStoreService", () => {
         ssmClientStub.calledWithMatch({
           Name: "/core-front/test-parameter",
-        }),
+        } as any),
```
This doesn't seem to be needed

DanCorderIPV commented on src/app/ipv/tests/handleJourneyActionRequest.test.ts:
```diff
@@ -2,161 +2,178 @@ import { expect } from "chai";
+    );
   });
 
   it("should postJourneyEvent and use ip address from header when not present in session", async function () {
```
This test and the one below are supposed to check that the correct IP address is being chosen - so both IP addresses should still be specified in the tests.
However, I can't see that the IP address is checked anywhere in the test, so perhaps these tests are obsolete?

DanCorderIPV replied:
Looks like Thea added the original test on 21/10/24. Could you please talk to her and see if these tests are doing what they are supposed to?

MikeCollingwood replied:
It looks like this was the last thought on the matter: https://github.com/govuk-one-login/ipv-core-front/pull/1611#discussion_r1809049591. This (and the beneath) test looks redundant as the logic occurs within the postJourneyEvent method, which has a test covering it:
`coreBackService.test.ts > "should postJourneyEvent with correct parameters and headers"`.

DanCorderIPV commented on src/app/ipv/tests/handleJourneyActionRequest.test.ts:
```diff
@@ -2,161 +2,178 @@ import { expect } from "chai";
 
-  it("should call redirect given 'contact' event", async function () {
-    const req = { ...testReq, body: { journey: "contact" } };
+  it("should use session IP if present", async function () {
```
Again there doesn't appear to be any testing of the IP address - so what is this test for?

MikeCollingwood committed:
PYIC-7329:
- Remove redundant handleJourneyActionRequest.test.ts ipAddress tests
- small refactors from PR comments

MikeCollingwood committed:
Merge branch 'main' into pyic-7329

MikeCollingwood committed:
PYIC-7329:
- Remove redundant handleJourneyActionRequest.test.ts ipAddress test
